backend/src/search/cmd.rs

```diff
@@ -104,6 +104,26 @@ macro_rules! info_line {
     };
 }
 
+macro_rules! with_index {
```

Comment:
It's a macro because passing async functions is pain :/ I mean I could probably do it ye, but it seemed like avoiding unnecessary headache :D 

And yes, it will be used in a later commit, as you probably have seen by now.